### PR TITLE
Implemented NgRx logic for LoginComponent

### PR DESCRIPTION
**Pull request by ipykalo**

### CODE REVIEW

File: src/app/modules/login/login/login.component.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import { Component, OnDestroy, OnInit } from '@angular/core';
 import { AbstractControl, FormBuilder, FormGroup, Validators } from '@angular/forms';
-import { LoginService } from '../login.service';
 import { messages } from '../constants/messages';
-import { Subject } from "rxjs";
 import { takeUntil } from "rxjs/operators";
 import { Router } from "@angular/router";
+import { Store } from "@ngrx/store";
+import { login } from "../../../store/actions/login.actions";
```
Comment (denis-starchenko) [inline src/app/modules/login/login/login.component.ts]: Please add short path for @actions

File: src/app/modules/users/groups/groups.component.ts [hunk 2]
```diff
@@ -34,8 +43,41 @@ export class GroupsComponent implements OnInit, OnDestroy {
...
+    const dialogRef = this.dialog.open(DialogAddGroupComponent, {
+      width: '400px',
+      data: this.groupData
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
+      const data = {
+        name: result.name,
+        description: result.description,
+        students_count: Number(result.students_count),
```
Comment (denis-starchenko) [inline src/app/modules/users/groups/groups.component.ts]: Please use input element with type `number`

File: src/app/store/effects/groups.effects.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        ofType(GroupActions.createGroup),
+        mergeMap(action =>
+          this.groupsService.createGroup({
+            name: action.name,
+            description: action.description,
+            students_count: action.students_count,
+            cost: {
+              sum: action.cost.sum,
+              currencyCode: action.cost.currencyCode
+            }
```
Comment (denis-starchenko) [inline src/app/store/effects/groups.effects.ts]: Its code can be simplified

File: src/app/store/index.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import {ActionReducerMap} from "@ngrx/store";
+import * as fromLayout from './reducers/login.reducers';
+
+export interface State {
+  login: fromLayout.State
+}
+
+export const reducers: ActionReducerMap<State> = {
+  login: fromLayout.reducer
+};
```
Comment (denis-starchenko) [inline src/app/store/index.ts]: Please move this file to `reducers` catalog and make reducer for `groups` actions

File: src/app/modules/login/login/login.component.ts (hunk 1 above)
Comment (ipykalo) [reply src/app/modules/login/login/login.component.ts]: Done

File: src/app/modules/users/groups/groups.component.ts (hunk 2 above)
Comment (ipykalo) [reply src/app/modules/users/groups/groups.component.ts]: Done

File: src/app/store/index.ts (hunk 4 above)
Comment (ipykalo) [reply src/app/store/index.ts]: Done

File: src/app/store/effects/groups.effects.ts (hunk 3 above)
Comment (ipykalo) [reply src/app/store/effects/groups.effects.ts]: Done

**Pull request merged**